pkg/package-server-manager/controller_test.go

```diff
@@ -133,83 +163,104 @@ func TestEnsureCSV(t *testing.T) {
 	singleReplicas := pointer.Int32(singleReplicaCount)
 	image := getImageFromManifest()
 
+	type wanted struct {
+		expectedBool bool
+		expectedErr  error
+	}
+
 	tt := []struct {
 		name            string
 		inputCSV        *olmv1alpha1.ClusterServiceVersion
 		expectedCSV     *olmv1alpha1.ClusterServiceVersion
 		highlyAvailable bool
-		want            bool
+		want            wanted
```

Comment:
all test cases seem to set expectedErr to nil. Is it worth exercising the failure path as well?
Seems like it only happens when there are issues deserializing the csv

Reply:
That's correct, all tests expect no error from this method. The request to add an error seems reasonable, but the only state where ensureCSV can return an error is when it has [an error unmarshalling the packageserver.yaml](https://github.com/openshift/operator-framework-olm/blob/2149aebcc71367e6fba8f5416374917dae1e6a1c/pkg/manifests/manifests.go#L18-L25), which is hardcoded and not exposed via the function. While we could mock the function or update it to take a configurable path, it seemed like overkill and I didn't want to increase the scope of this PR given that it will need to be backported to 4.9.

If you feel strongly that we should test this path, I can update it when I'm back from vacation next week.

Reply:
my preference would be to not increase the scope of this PR either, specially with the need to backport in mind. We could always have a follow up PR for the test